src/FilteredActionList/FilteredActionList.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import {GroupedListProps, ListPropsBase} from '../ActionList/List'
+import TextInput from '../TextInput'
+import Box from '../Box'
+import SelectMenu from '../SelectMenu'
+import {ActionList} from '../ActionList'
+
+export interface FilteredActionListProps extends Partial<Omit<GroupedListProps, keyof ListPropsBase>>, ListPropsBase {
+  loading?: boolean
+  placeholderText: string
+  filterValue: string
+  onChange: (e: React.ChangeEvent<HTMLInputElement>) => unknown
```

Comment:
```suggestion
  onChange: (filterValue: string, event: React.ChangeEvent<HTMLInputElement>) => unknown
```
Suggestion: Pass both the new filter value and the event.